### ISSUE TITLE
path too long

### ISSUE THREAD

**Issue by DarkMetalMouse** [ktheod/OneDriveBully#21]
I get this error frequently when OneDriveBully is running. It has never worked.
![image](https://user-images.githubusercontent.com/51059131/119236902-a9722d80-bb42-11eb-8a43-d8e502a61c94.png)

**Comment by DarkMetalMouse** [comment]
Here are the details:

```
See the end of this message for details on invoking 
just-in-time (JIT) debugging instead of this dialog box.

************** Exception Text **************
System.IO.PathTooLongException: The specified path, file name, or both are too long. The fully qualified file name must be less than 260 characters, and the directory name must be less than 248 characters.
   at System.IO.PathHelper.GetFullPathName()
   at System.IO.Path.LegacyNormalizePath(String path, Boolean fullCheck, Int32 maxPathLength, Boolean expandShortPaths)
   at System.IO.Path.GetFullPathInternal(String path)
   at System.IO.FileInfo.Init(String fileName, Boolean checkHost)
   at OneDriveBully.MyFunctions.IsSymbolic(String path)
   at OneDriveBully.MyFunctions.getOneDriveForSymLinks()
   at OneDriveBully.SettingsForm.Refresh_dgv()
   at System.Windows.Forms.Control.OnClick(EventArgs e)
   at System.Windows.Forms.Button.OnClick(EventArgs e)
   at System.Windows.Forms.Button.OnMouseUp(MouseEventArgs mevent)
   at System.Windows.Forms.Control.WmMouseUp(Message& m, MouseButtons button, Int32 clicks)
   at System.Windows.Forms.Control.WndProc(Message& m)
   at System.Windows.Forms.ButtonBase.WndProc(Message& m)
   at System.Windows.Forms.Button.WndProc(Message& m)
   at System.Windows.Forms.NativeWindow.Callback(IntPtr hWnd, Int32 msg, IntPtr wparam, IntPtr lparam)


************** Loaded Assemblies **************
mscorlib
    Assembly Version: 4.0.0.0
    Win32 Version: 4.8.4341.0 built by: NET48REL1LAST_C
    CodeBase: file:///C:/Windows/Microsoft.NET/Framework64/v4.0.30319/mscorlib.dll
----------------------------------------
OneDriveBully
    Assembly Version: 1.3.0.0
    Win32 Version: 1.3.0.0
    CodeBase: file:///D:/Programs/OneDriveBully/OneDriveBully.exe
----------------------------------------
System.Windows.Forms
    Assembly Version: 4.0.0.0
    Win32 Version: 4.8.4341.0 built by: NET48REL1LAST_C
    CodeBase: file:///C:/WINDOWS/Microsoft.Net/assembly/GAC_MSIL/System.Windows.Forms/v4.0_4.0.0.0__b77a5c561934e089/System.Windows.Forms.dll
----------------------------------------
System
    Assembly Version: 4.0.0.0
    Win32 Version: 4.8.4300.0 built by: NET48REL1LAST_C
    CodeBase: file:///C:/WINDOWS/Microsoft.Net/assembly/GAC_MSIL/System/v4.0_4.0.0.0__b77a5c561934e089/System.dll
----------------------------------------
System.Drawing
    Assembly Version: 4.0.0.0
    Win32 Version: 4.8.4084.0 built by: NET48REL1
    CodeBase: file:///C:/WINDOWS/Microsoft.Net/assembly/GAC_MSIL/System.Drawing/v4.0_4.0.0.0__b03f5f7f11d50a3a/System.Drawing.dll
----------------------------------------
System.Configuration
    Assembly Version: 4.0.0.0
    Win32 Version: 4.8.4190.0 built by: NET48REL1LAST_B
    CodeBase: file:///C:/WINDOWS/Microsoft.Net/assembly/GAC_MSIL/System.Configuration/v4.0_4.0.0.0__b03f5f7f11d50a3a/System.Configuration.dll
----------------------------------------
System.Core
    Assembly Version: 4.0.0.0
    Win32 Version: 4.8.4341.0 built by: NET48REL1LAST_C
    CodeBase: file:///C:/WINDOWS/Microsoft.Net/assembly/GAC_MSIL/System.Core/v4.0_4.0.0.0__b77a5c561934e089/System.Core.dll
----------------------------------------
System.Xml
    Assembly Version: 4.0.0.0
    Win32 Version: 4.8.4084.0 built by: NET48REL1
    CodeBase: file:///C:/WINDOWS/Microsoft.Net/assembly/GAC_MSIL/System.Xml/v4.0_4.0.0.0__b77a5c561934e089/System.Xml.dll
----------------------------------------
System.Data
    Assembly Version: 4.0.0.0
    Win32 Version: 4.8.4270.0 built by: NET48REL1LAST_C
    CodeBase: file:///C:/WINDOWS/Microsoft.Net/assembly/GAC_64/System.Data/v4.0_4.0.0.0__b77a5c561934e089/System.Data.dll
----------------------------------------
System.Numerics
    Assembly Version: 4.0.0.0
    Win32 Version: 4.8.4084.0 built by: NET48REL1
    CodeBase: file:///C:/WINDOWS/Microsoft.Net/assembly/GAC_MSIL/System.Numerics/v4.0_4.0.0.0__b77a5c561934e089/System.Numerics.dll
----------------------------------------

************** JIT Debugging **************
To enable just-in-time (JIT) debugging, the .config file for this
application or computer (machine.config) must have the
jitDebugging value set in the system.windows.forms section.
The application must also be compiled with debugging
enabled.

For example:

<configuration>
    <system.windows.forms jitDebugging="true" />
</configuration>

When JIT debugging is enabled, any unhandled exception
will be sent to the JIT debugger registered on the computer
rather than be handled by this dialog box.


```

**Comment by NoMagicCode** [comment]
I got the exact same "Unhandled exception" error message while using v1.3. I've been using OneDriveBully for years -- it has worked flawlessly. I recently went to v1.3 and it was working flawlessly too, until I used the built-in capability to create the symbolic links (rather than using admin cmd line mklink). I probably did something wrong and possibly created a long path in an unexpected place. I ran a .NET repair tool and scanned the file system (SFC /SCANNOW, which said it repaired some corrupt files). Rebooting and reinstalling OneDriveBully made no difference; I continued to get the error message every time I started v1.3. I successfully went back to v1.2 (and mklink).

**Comment by Lordvader81** [comment]
Same issue here ....

It would be nice if the Dev would respond to these issues and give us a solution. I can't believe that in March of 2023 this is still a problem...😥

**Comment by teknolai** [comment]
For what it's worth, you can find the settings in `%LOCALAPPDATA%\K.Theodoridis\` and there is one `user.config`-file in the OneDriveBully folder. 
(E.g. `"C:\Users\Username\AppData\Local\K.Theodoridis\OneDriveBully.exe_Url_igc0i1hr2axmwy3biv45qgfg02ob5ho3\1.3.0.0\user.config"` ) 

If you rename/delete the existing file, you can add new settings.

Still doesn't fix the problem with the Unhandled exception, though.

**Comment by ktheod** [comment]
Hi all, thanks for the ticket and your comments. Apologies for the really long delay. 
This should be fixed in version 1.4 coming out this weekend.
Thanks,

**Comment by techflick** [comment]
Thank you very much!

31.05.2024 15:19:27 Cafeine ***@***.***>:
